scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,141 @@
+"""Module to generate a column map to their corresponding statistical variable defintions."""
+
+import os
+import json
+
+import pandas as pd
+
+#TODO:Remove this import and make it from ~/data/utils/ directory when it is available
+os.system(
+    'curl https://raw.githubusercontent.com/Abilityguy/data/dcid_gen/util/statvar_dcid_generator.py -LO statvar_dcid_generator.py'
+)
+from statvar_dcid_generator import get_stat_var_dcid
+
+
+
+def generate(specDict: dict, columnList: list) -> dict:
+    """Wrapper function for generateColMapBase class to generate column map.
+
+  Args:
+    specDict:
+      A dictionary containing specifications for the different properties of the statistical variable.
+    columnList:
+      A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+
+  Returns:
+    A dict mapping each column to their respective stat_var node definitions.
+  """
+    col_map_obj = GenerateColMapBase(specDict=specDict, columnList=columnList)
+    return col_map_obj._generate_stat_vars_from_spec()
+
+
+class GenerateColMapBase:
+    """module to generate a column map given a list of columns of the dataset and a JSON Spec
+
+  Attributes:
+    specDict: A dictionary containing specifications for the different properties of the statistical variable.
+    columnList: A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+  """
+
+    def __init__(self, specDict=None, columnList=None):
+        """module init """
+        self.features = specDict
+        self.column_list = columnList
+        self.column_map = {}
+
+    def _generate_stat_vars_from_spec(self):
+        """generates stat_var nodes for each column in column list and returns a dictionary of columns with their respective stat_var nodes"""
+        # for each column generate the definition of their respective statistical variable node
+        for col in self.column_list:
+            if 'ignoreColumns' in self.features and col not in self.features[
+                    'ignoreColumns']:
+                self.column_map[col] = self._column_to_statvar(col)
+        #TODO: Enable validating the generate columnMap: _valid_col_map():
+        return self.column_map
+
+    def _column_to_statvar(self, column):
+        """generates a dictionary statistical variable with all properties specified in the JSON spec for a single column"""
+        measurement_assigned = False
+        stat_var = {}
+        # part_list contains a list of tokens of the column name after splitting the
+        # column name based on the '!!' delimiter
+        part_list = column.split('!!')
```

Comment:
Can the delimiter be a configurable variable defaulting to '!!' in the constructor?